src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,288 @@
+    {
+        appwin->currentWindow()->on_clearMessagesButton_clicked();
+        return true;
+    }
+    if (wantExit(cmd))
+    {
+        appwin->on_actionQuit_triggered();
+        return true;
+    }
+    // return false indicating that custom command handler failed to handle the command
+    return false;
+}
+
+QPair<bool, QString> FakeVimCommand::wantNewFile(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "new")
+    {
+        if (ex.args == "c++" || ex.args == "cpp")
+            return {true, "C++"};
+        else if (ex.args == "py" || ex.args == "python")
+            return {true, "Python"};
+        else if (ex.args == "java")
+            return {true, "Java"};
+        else if (ex.args.isEmpty())
+            return {true, SettingsHelper::getDefaultLanguage()};
+        else
+            SHOW_STATUS_MESSAGE(
+                tr("Error: %1 requires an empty or one of \"cpp\", \"java\" and \"python\" argument").arg(ex.cmd));
+    }
+
+    return {false, ""};
+}
+
+QPair<bool, QString> FakeVimCommand::wantOpenFile(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "open" || ex.cmd == "opn")
+    {
+        if (ex.args.isEmpty())
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 requires a file path").arg(ex.cmd));
+            return {true, ""};
+        }
+
+        QFileInfo file(ex.args);
+        if (!file.exists())
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 does not exists").arg(ex.args));
+            return {true, ""};
+        }
+        else if (!file.isFile())
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 is not a file").arg(ex.args));
+            return {true, ""};
+        }
+
+        else if (!Util::cppSuffix.contains(file.suffix()) && !Util::javaSuffix.contains(file.suffix()) &&
+                 !Util::javaSuffix.contains(file.suffix()))
+        {
+            SHOW_STATUS_MESSAGE(tr("Error: %1 is not a source file of C/C++, Java or Python").arg(ex.args));
+            return {true, ""};
+        }
+        else
+        {
+            return {true, ex.args};
+        }
+    }
+    return {false, ""};
+}
+
+bool FakeVimCommand::wantCompile(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "compile" || ex.cmd == "cmp";
+}
+
+bool FakeVimCommand::wantCompileRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "crun" || ex.cmd == "crn";
+}
+
+bool FakeVimCommand::wantDetachedRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "drun" || ex.cmd == "drn";
+}
+
+QPair<bool, int> FakeVimCommand::wantRun(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "run")
+    {
+        if (ex.args.isEmpty()) // run all testcases
+        {
+            return {true, 0};
+        }
+        bool ok = false;
+        int caseNum = ex.args.toInt(&ok);
+        if (ok)
+            return {true, caseNum};
+        else
+        {
+            SHOW_STATUS_MESSAGE(tr("%1 is not a number").arg(caseNum));
```

Comment:
I will solve it by making fakevimhandler shareable across different tabs. So YANK in one tab can PASTE in other and so on and so forth. It also needs some code refactoring on which I will solve this.

Reply:
> YANK in one tab can PASTE in other and so on and so forth.

This can already be achieved even if we use multiple different handlers. They all share the same YANK history.

As for the commands, I am refactoring the FakeVimCommands file completely.